array_experimental_test.go

```diff
@@ -131,3 +138,62 @@ func TestGetConsolidationPlan(t *testing.T) {
 
 	checkConsolidationPlan(t, cplan)
 }
+
+func TestConsolidateFragments(t *testing.T) {
+	// The test is skipped pending a core release for 2.25.0 that includes this fix:
+	// https://github.com/TileDB-Inc/TileDB/pull/5135
+	t.Skip("Skipping fragment list consolidation SC-51140")
+
+	array := createTestArray(t)
+
+	numFrags := uint32(5)
+	for i := uint32(0); i < numFrags; i++ {
+		writeTestArray(t, array, []int32{0, 1, 2, 3, 4, 5, 6, 7, 8, 9})
+	}
+
+	fragmentInfo, err := NewFragmentInfo(array.context, array.uri)
+	require.NoError(t, err)
+
+	err = fragmentInfo.Load()
+	require.NoError(t, err)
+
+	fragInfoNum, err := fragmentInfo.GetFragmentNum()
+	require.NoError(t, err)
+	require.Equal(t, numFrags, fragInfoNum)
+	fragUris := make([]string, numFrags)
+	for i := uint32(0); i < numFrags; i++ {
+		uri, err := fragmentInfo.GetFragmentURI(i)
+		require.NoError(t, err)
+		fragUris[i] = uri
+	}
+
+	// Default consolidation mode is 'fragments'.
+	config, err := array.context.Config()
+	require.NoError(t, err)
+
+	err = array.ConsolidateFragments(config, fragUris)
+	require.NoError(t, err)
+
+	// Check that the new consolidated fragment was created.
+	err = fragmentInfo.Load()
+	require.NoError(t, err)
+	fragInfoNum, err = fragmentInfo.GetFragmentNum()
+	require.NoError(t, err)
+	fragToVacuumNum, err := fragmentInfo.GetToVacuumNum()
+	require.NoError(t, err)
+	require.Equal(t, numFrags, fragToVacuumNum)
```

Comment:
…but instead, you can write `require.EqualValues(t, numFrags, fragToVacuumNum)` (or `require.Equal(t, uint32(numFrags), ...)`) to avoid having a bunch of `uint32(n)`s above.

Reply:
Much better, very nice :+1: